raiden/utils/migrations/v17_to_v18.py

```diff
@@ -0,0 +1,88 @@
+import json
+
+from raiden.exceptions import ChannelNotFound
+from raiden.storage.sqlite import SQLiteStorage
+from raiden.transfer.state import RouteState
+from raiden.utils.typing import Any, Dict
+
+SOURCE_VERSION = 17
+TARGET_VERSION = 18
+
+
+def get_token_network_by_identifier(snapshot, token_network_identifier):
+    identifiers_to_paymentnetworks = snapshot['identifiers_to_paymentnetworks']
+    for paymentnetwork in identifiers_to_paymentnetworks.values():
+        for token_network in paymentnetwork['tokennetworks']:
+            if token_network['address'] == token_network_identifier:
+                return token_network
+    return None
+
+
+def _transform_snapshot(raw_snapshot: Dict[Any, Any]):
+    """
+    This migration upgrades the object:
+    - `MediatorTransferState` such that a list of routes is added
+    to the state to be able to route a waiting transfer in case the
+    receiving node comes back online.
+    """
+    snapshot = json.loads(raw_snapshot)
+    secrethash_to_task = snapshot['payment_mapping']['secrethashes_to_task']
+    for task in secrethash_to_task.values():
+        if task['_type'] != 'raiden.transfer.state.MediatorTask':
+            continue
+
+        mediator_state = task.get('mediator_state')
+
+        # The migration should be idempotent.
+        if 'routes' in mediator_state:
```

Comment:
Hmm shouldn't this be an assertion? If somehow we end up in the `v17_to_v18` migration and a mediator state already has the `routes` attribute something is really really wrong. I am not sure if ignoring is the best option.

Reply:
Yes, after merging my PR which passes the correct `old_version`, an assert should be safe enough.